src/main/java/webApp/webController.java

```diff
@@ -67,13 +67,13 @@ public Response createSurvey(@RequestBody SurveyMessage surveyMessage) {
         HashSet<String> set = new HashSet<>(questionTexts);
 
         if(set.size() < questionTexts.size()){
-            return new Response(null, "error", "Duplicate questions detected");
+            return new Response(null, null, "error", "Duplicate questions detected");
         }
 
         Survey survey = new Survey(surveyMessage.getName());
         survey.setQuestions(questionList);
         repo.save(survey);
-        return new Response(survey.getId(), "ok", "done");
+        return new Response(survey.getId(), survey.getLink(), "ok", "done");
```

Comment:
This is the only place we need to test getting a link because the link is not and should not be returned in other cases.